src/Raven.Server/Documents/Indexes/IndexDefinitionBaseServerSide.cs

```diff
@@ -171,9 +175,10 @@ protected IndexDefinitionBase(string name, IEnumerable<string> collections, Inde
             Priority = priority;
             State = state;
             _indexVersion = indexVersion;
+            ClusterIndex.ClusterIndexForState = clusterIndexForState ?? 0;
```

Comment:
I see that 0 is some default value here? Noticed it in few places. If yes then maybe this field should not be nullable?